pkg/ctl/create/addon.go

```diff
@@ -92,7 +92,20 @@ func createAddonCmd(cmd *cmdutils.Cmd) {
 			return err
 		}
 
+		summaries, err := addonManager.GetAll(ctx)
+		if err != nil {
+			return err
+		}
+		activeAddons := make(map[string]struct{}, len(summaries))
+		for _, s := range summaries {
+			activeAddons[s.Name] = struct{}{}
+		}
+
 		for _, a := range cmd.ClusterConfig.Addons {
+			if _, alreadyPresent := activeAddons[a.Name]; alreadyPresent {
+				logger.Info("Addon %s is already present in this cluster, and won't be recreated", a.Name)
+				continue
+			}
```

Comment:
should we also update/improve unit tests around this?

Reply:
There's no unit test related to this command ATM. I'd suggest adding a tech debt ticket?